src/QMCWaveFunctions/Fermion/SlaterDetBuilder.h

```diff
@@ -124,7 +119,7 @@ class SlaterDetBuilder : public WaveFunctionComponentBuilder
       extVar = "Coeff_" + std::to_string(ext_level);
 
     if (!hin.readEntry(ci_coeff, extVar))
-      APP_ABORT("Could not read CI coefficients from HDF5");
+      throw std::runtime_error("Could not read CI coefficients from HDF5");
```

Comment:
Unless I'm missing something this appears to get called on all ranks.  Since I'd expect them to all fail this should be a `UniformCommunicateError` this should be our way to signal an error is expected on all ranks so we should wait for all of them to fail and report the error on the head rank.

Reply:
I'm surprised to see this is called from all ranks. A fix will be on the caller side.
Reading redundant 1M coefs from all the ranks is not good.

Reply:
Perhaps a topic for an issue / subsequent PR? To me it looks like the SlaterDetBuilder hasn't been through cleanup with a view of potentially having lots of MPI ranks. Also, there is functionality in the .h that has no business not being in a .cpp.

Reply:
Alright this needs to be revisited on a couple of fronts and isn't something that can be solved in this PR.  I'm going to approve.

Reply:
(If updates are going to wait, the issue should be created ahead of merge)

Reply:
I've made issues for the UCE related stuff.  The reading a million coefficients on every rank issue is separate and not added.